observer/bin/spark-observer.js

```diff
@@ -62,9 +48,23 @@ await pgPool.query('SELECT 1')
 
 console.log('Listening for impact evaluator events')
 
-ieContract.on('Transfer', (to, amount, ...args) => {
-  /** @type {number} */
-  const blockNumber = args.pop()
-  console.log('Transfer %s FIL to %s at epoch %s', amount, to, blockNumber)
-  // TODO: update the database
-})
+// Get the last block we checked. Even though there should be only one row, use MAX just to be safe
+const lastCheckedBlock = await pgPool.query(
+  'SELECT MAX(last_block) AS last_block FROM reward_transfer_last_block'
+).then(res => res.rows[0].last_block)
+
+// Listen for Transfer events from the IE contract
+ieContract.queryFilter(ieContract.filters.Transfer(), lastCheckedBlock)
+  .then(events => {
+    for (const event of events) {
+      console.log('%s FIL to %s at block %s', event.args.amount, event.args.to, event.blockNumber)
+      updateDailyFilStats(
+        pgPool,
+        {
+          to_address: event.args.to,
+          amount: event.args.amount,
+          blockNumber: event.blockNumber
+        }
+      )
+    }
+  })
```

Comment:
Also, because I made the switch to use queryFilter instead of "on", this is now a one-time call, instead of a continuously listening action, which is wrong as-is.

We need to either run this script on a schedule (e.g. once daily), or we need to run a schedule within this script's execution)

Reply:
I think we can do cron jobs using Fly IO config. https://fly.io/docs/machines/flyctl/fly-machine-run/#start-a-machine-on-a-schedule

 What do we think about that?  @juliangruber @bajtos

Reply:
I'd prefer to run something like

```js
while (true) {
  await run()
  await sleep(ONE_HOUR)
}
```

This way we use less platform features, and own the iteration logic